R/readWrite.R

```diff
@@ -0,0 +1,97 @@
+    gene_countsfn <- paste(path,"counts_gene.txt",sep="")
+    write.table(gene_counts, file= gene_countsfn, sep="\t", quote=FALSE)
+  }
+}
+#' Write annotation GRangesList into a GTF file
+#' @title write GRangeslist into GTF file
+#' @param annotation a \code{\link{GRangesList}} object
+#' @param file the output gtf file name
+#' @param geneIDs an optional dataframe of geneIDs (column 2) with the corresponding transcriptIDs (column 1)
+#' @return gtf a GTF dataframe
+#' @export
+writeToGTF <- function (annotation,file,geneIDs=NULL) {
+  if (missing(annotation) | missing(file)){
+    stop('Both GRangesList and the name of file are required.')
+  }else if (class(annotation) != "CompressedGRangesList"){
+    stop('The inputted GRangesList is of the wrong class.')
+  }else {
+    df <- as.data.frame(annotation)
+    df$exon_rank <- paste('exon_number "',df$exon_rank,'";',sep= '')
+    if (missing(geneIDs)){
+      if (!is.null(mcols(annotation, use.names=FALSE)$GENEID)){
+        geneIDs <- data.frame(mcols(annotation, use.names=FALSE)$TXNAME,mcols(annotation, use.names=FALSE)$GENEID)
+        colnames(geneIDs) <- c("TXNAME","GENEID")
+      }
+    }
+    df <- merge(df,geneIDs,by.x="group_name",by.y="TXNAME",all=TRUE)
+    df$group_name <- paste('transcript_id "',df$group_name,'";',sep= '')
+    df$GENEID <- paste('gene_id "',df$GENEID,'";',sep= '')
+    gtf_exon <- data.frame(seqname=df$seqnames, source= "Bambu",feature= "exon",
+                          start=df$start,end=df$end,score=".",strand=df$strand,frame=".",
+                          attributes= paste(df$GENEID,df$group_name,df$exon_rank))
+    df_end <- df[order(df$end, decreasing = TRUE),]
+    df_end <- data.frame(group_name=df_end$group_name,uend=df_end$end)
+    df_end <- df_end[!duplicated(df_end$group_name),]
+    df <- df[!duplicated(df$group_name),]
+    df <- merge(df,df_end,by="group_name",all=TRUE)
+    gtf_trns <- data.frame(seqname=df$seqnames, source= "Bambu",feature= "transcript",
+                             start=df$start,end=df$uend,score=".",strand=df$strand,frame=".",
+                             attributes= paste(df$GENEID,df$group_name))
+    }
+    gtf <- rbind(gtf_trns,gtf_exon)
+    gtf <- gtf[order(gtf$attributes),]
+    write.table(gtf, file= file, quote=FALSE, row.names=FALSE, col.names=FALSE, sep = "\t")
+} 
+
+
+#' Outputs GRangesList object from reading a GTF file
+#' @title convert a GTF file into a GRangesList
+#' @param file a \code{\link{.gtf}} file
+#' @return grlist a \code{\link{GRangesList}} object, with two columns
+#' \itemize{
+#'   \item TXNAME specifying prefix for new gene Ids (genePrefix.number), defaults to empty
+#'   \item GENEID indicating whether filter to remove read classes which are a subset of known transcripts(), defaults to TRUE
+#'   }
+#' @export
+readFromGTF <- function(file){
+  if (missing(file)){
+    stop('A GTF file is required.')
+  }else{
+    data=read.delim(file,header=FALSE)
```

Comment:
```suggestion
    data=read.delim(file,header=FALSE, comment.char='#')
```
(do we need to remove # lines for gtf?)